language/move-lang/src/expansion/translate.rs

```diff
@@ -384,13 +403,27 @@ fn struct_def(
         resource_opt,
         type_parameters: pty_params,
         fields: pfields,
+        invariants: pinvariants,
     } = pstruct;
     let type_parameters = type_parameters(context, pty_params);
     let fields = struct_fields(context, &name, pfields);
+    let invariants = pinvariants
+        .into_iter()
+        .map(|sp!(loc, P::Invariant_ { kind, exp: pexp })| {
+            sp(
+                loc,
+                E::Invariant_ {
+                    kind,
```

Comment:
Same as above. Use punning, and it should fit on less lines and be easier to read

Reply:
Done